src/allmydata/immutable/upload.py

```diff
@@ -91,7 +91,7 @@ def _serialize_existing_shares(existing_shares):
 
 def _serialize_happiness_mappings(happiness_mappings):
     return {
-        sharenum: base32.b2a(serverid)
+        str(sharenum): ensure_str(base32.b2a(serverid))
```

Comment:
`str` -> `ensure_str` ...? (or, why not?

Reply:
```
>>> b = b"abc"
>>> str(b)
"b'abc'"
```

Reply:
sorry, I guess I wasn't clear: why `str(sharenum)` and not `ensure_str(sharenum)`...?

Reply:
sharenum is an int so it's just a matter of making sure keys are strings, vs dealing with bytes.